### DISCUSSION TITLE
How about CRUD?

### DISCUSSION THREAD

**Discussion by tdkehoe** [angular/angularfire#2946] [General]
The "Quickstart" tutorial shows how to set up Angular to read from a Firestore database. Can we also get instructions for Create, Update, and Delete?

There are a zillion blogs that teach how to make a CRUD app with AngularFire and Material. Every one starts with a couple hundred lines of Material to make your to-do list look beautiful. This code is always broken so I spend hours trying to update old Material code. Eventually at the end of the tutorial you learn how to add AngularFire to your app. This code is also always broken. Angular, AngularFire, and Material are all moving targets so unless a tutorial is updated regularly something will be broken.

Will someone please explain how to use AngularFire to Create, Read, Update, and Delete a Firestore database from an Angular frontend? No styling! No CSS! No Material! I don't care what it looks like! I just want CRUD operations. I'll style it later.

Sorry about the rant but I've wasted !@#$ hours on broken blog posts. Why doesn't AngularFire have official documentation for CRUD? And why do blog posts always start with hours of styling? Do developers start with styling? I don't. I start with the CRUD operations and leave styling for later.

Apologies again for the rant. :-)

**Comment by jamesdaniels** [comment]
Great thing about the new API (documentation pending) is it's the same API as the vanilla Firebase JS SDK (modular) with a few convenience additions. So you can look [at the official Firebase docs](https://firebase.google.com/docs/firestore/manage-data/add-data), just change your import paths from `firebase/*` to `@angular/fire/*`, and everything works.

**Reply by tdkehoe** [reply]
When can we expect the documentation on the new API?

**Reply by jamesdaniels** [reply]
> When can we expect the documentation on the new API?

I'm pretty busy filling the api gaps & fixing issues that are popping up but will take a stab at the new docs next week. welcome any help.

**Reply by tdkehoe** [reply]
I'm happy to help and I used to work as a tech writer but I don't know how much help I would be. I'm an expert with AngularJS and Firebase and I can handle Material. I was doing fairly well with Angular 11 but I couldn't get Firebase hooked up. Now with Angular 12 I just get error messages about '@angular-devkit' and nothing can fix this.

**Comment by tdkehoe** [comment]
I wrote a tutorial that shows how to use every feature of Firestore with AngularFire: https://github.com/tdkehoe/Using-Firebase-with-Angular-and-AngularFire
